src/util/fs.js

```diff
@@ -6,34 +6,47 @@ import * as promise from './promise.js';
 import {promisify} from './promise.js';
 import map from './map.js';
 
+const userHome = require('./user-home-dir').default;
 const fs = require('fs');
 const globModule = require('glob');
 const os = require('os');
 const path = require('path');
 
 export const lockQueue = new BlockingQueue('fs lock');
 
-export const readFileBuffer = promisify(fs.readFile);
-export const writeFile: (path: string, data: string) => Promise<void> = promisify(fs.writeFile);
-export const readlink: (path: string, opts: void) => Promise<string> = promisify(fs.readlink);
-export const realpath: (path: string, opts: void) => Promise<string> = promisify(fs.realpath);
-export const readdir: (path: string, opts: void) => Promise<Array<string>> = promisify(fs.readdir);
-export const rename: (oldPath: string, newPath: string) => Promise<void> = promisify(fs.rename);
-export const access: (path: string, mode?: number) => Promise<void> = promisify(fs.access);
-export const stat: (path: string) => Promise<fs.Stats> = promisify(fs.stat);
-export const unlink: (path: string) => Promise<void> = promisify(require('rimraf'));
-export const mkdirp: (path: string) => Promise<void> = promisify(require('mkdirp'));
-export const exists: (path: string) => Promise<boolean> = promisify(fs.exists, true);
-export const lstat: (path: string) => Promise<fs.Stats> = promisify(fs.lstat);
-export const chmod: (path: string, mode: number | string) => Promise<void> = promisify(fs.chmod);
-export const link: (path: string) => Promise<fs.Stats> = promisify(fs.link);
-export const glob: (path: string) => Promise<Array<string>> = promisify(globModule);
+export const createWriteStream = (path: string, options?: Object): fs.WriteStream =>
+  fs.createWriteStream(expandPath(path), options);
+export const createReadStream = (path: string, options?: Object): fs.ReadStream =>
+  fs.createReadStream(expandPath(path), options);
+export const writeFileSync = (path: string, data: Buffer | string, options?: Object | string): void =>
+  fs.writeFileSync(expandPath(path), data, options);
+export const readFileSync = (path: string, options?: Object | string): any =>
+  fs.readFileSync(expandPath(path), options);
+export const existsSync = (path: string): boolean => fs.existsSync(expandPath(path));
```

Comment:
I think we should not use util/fs.js as a proxy to all methods to Node/fs module.

I would add more code to support.
Also I would not want *Sync methods to be easily exposed because people sometimes are tempted to use them instead of async/await.